### PR TITLE
Ensure that Collection[Visualization] usage vars are typed correctly

### PR DESCRIPTION
**Pull request by Oddant1**
Ensures that usage vars representing Collection[Visualization] are typed correctly. Previously they were typed as just visualization.

### CODE REVIEW

**Comment by gregcaporaso** [comment]
Thanks @Oddant1 - is there a test we could add for this? And did you confirm that it works with q2-boots to resolve the issue I shared?

**Comment by Oddant1** [comment]
blocking qiime2/q2cli#343

**Comment by Oddant1** [comment]
@gregcaporaso I did make sure it resolved the specific issue in boots yes. The problem was that we were mislabeling `Collection[Visualization]` as just `Visualization` due to how the if statements were ordered. I also found some other issues with how the usage API was handling collections and cleaned those up which made this a bit bigger than originally intended.

Additionally there is another PR qiime2/q2cli#343 that came directly out of this one to test that the cli is handling this properly. I resolved some issues I found over there as well.

If these two are now more than you want to review, just let me know. For now I will assign you, but like I said they ended up being a bit more than just that specific issue.

**Comment by gregcaporaso** [comment]
Thanks @Oddant1! I'm going to unassign myself for now, and we can discuss who will review this one in the engineering meeting tomorrow (maybe it'll come back to me, but let's see what everyone's availability is).

File: qiime2/core/testing/examples.py [hunk 1]
```diff
@@ -264,6 +264,24 @@ def collection_dict_of_ints(use):
...
+
+def collection_of_visualizations(use):
+    ints = use.init_artifact('ints', ints1_factory)
+
+    vizualizations, = use.action(
+        use.UsageAction(plugin_id='dummy_plugin',
+                        action_id='viz_collection_pipeline'),
+        use.UsageInputs(
+            ints=ints
+            ),
```
Comment (colinvwood) [inline qiime2/core/testing/examples.py]:
```suggestion
        use.UsageInputs(
            ints=ints
        ),
```

File: qiime2/tests/test_artifact_api.py [hunk 2]
```diff
@@ -278,14 +278,40 @@ def test_optional_inputs(self):
...
+    raise AssertionError
+hits = sorted(out_artifact_collection['Foo']._archiver.data_dir.glob('file1.txt'))
+if len(hits) != 1:
+    raise ValueError
+target = hits[0].read_text()
+match = re.search('1', target, flags=re.MULTILINE)
+if match is None:
+    raise AssertionError"""  # noqa: E501
+
+        self.maxDiff = None
```
Comment (colinvwood) [inline qiime2/tests/test_artifact_api.py]:
```suggestion
```

**Review by colinvwood — APPROVED** [review]

**Review by colinvwood — APPROVED** [review]

**Comment by gregcaporaso** [comment]
Thanks @Oddant1 and @colinvwood!

**Pull request merged**